molecule/command/init.py

```diff
@@ -42,88 +40,82 @@ class Init(base.Base):
-
-            self.clean_meta_main(role_path)
-
-        env = jinja2.Environment(
-            loader=jinja2.PackageLoader('molecule', 'template'),
-            keep_trailing_newline=True)
-
-        t_molecule = env.get_template(self.molecule.config.config['molecule'][
-            'init']['templates']['molecule'])
-        t_playbook = env.get_template(self.molecule.config.config['molecule'][
-            'init']['templates']['playbook'])
-        t_test_default = env.get_template(self.molecule.config.config[
-            'molecule']['init']['templates']['test_default'])
-
-        if (self.molecule._args['--docker']):
-            t_molecule = env.get_template(self.molecule.config.config[
-                'molecule']['init']['templates']['molecule_docker'])
-        if (self.molecule._args['--openstack']):
-            t_molecule = env.get_template(self.molecule.config.config[
-                'molecule']['init']['templates']['molecule_openstack'])
-
-        sanitized_role = re.sub('[._]', '-', role)
-        with open(
-                os.path.join(role_path, self.molecule.config.molecule_file),
-                'w') as f:
-            f.write(t_molecule.render(config=self.molecule.config.config,
-                                      role=sanitized_role))
-
-        with open(
-                os.path.join(
-                    role_path,
-                    self.molecule.config.config['ansible']['playbook']),
-                'w') as f:
-            f.write(t_playbook.render(role=role))
-
-        testinfra_path = os.path.join(
-            role_path,
-            self.molecule.config.config['molecule']['testinfra_dir'])
-
-        if not os.path.isdir(testinfra_path):
-            os.mkdir(testinfra_path)
-
-        with open(os.path.join(testinfra_path, 'test_default.py'), 'w') as f:
-            f.write(t_test_default.render())
-
-        msg = 'Successfully initialized new role in {}'
-        util.print_success(msg.format(role_path))
+        msg = 'Successfully initialized new role in {}...'
+        util.print_success(msg.format(os.path.join(role_path, role)))
         util.sysexit(0)
+
+    def _init_existing_role(self, role, role_path):
+        driver = self._get_driver()
+        extra_context = self._get_cookiecutter_context(role, driver)
+
+        util.print_info("Initializing molecule in current directory...")
+        for template in ['playbook', 'driver/{}'.format(driver)]:
+            self._create_template(template, extra_context, role_path)
+
+    def _init_new_role(self, role, role_path):
+        driver = self._get_driver()
+        extra_context = self._get_cookiecutter_context(role, driver)
+
+        util.print_info("Initializing role {}...".format(role))
+        for template in ['galaxy_init', 'playbook', 'driver/{}'.format(driver),
+                         'verifier/testinfra', 'verifier/serverspec']:
+            self._create_template(template, extra_context, role_path)
```

Comment:
I've probably implemented cookiecutter somewhat oddly here.  Wanted modular templates, since we need to write an entire role tree with modular test suites.  Other times we only want a couple files from the templates.

I figure we can iterate on this, and make it better as we learn more about cookiecutter. 🍪